### PR TITLE
"syscon fix" 

### PR DESCRIPTION
**Pull request by kjhayes**
It's not exactly a "fix" it's just if we tried to run on anything other than the exact QEMU setup in run.sh the "syscon" driver would fail.

syscon is just a generic set of registers but we had hardcoded access offsets and values which I'm guessing were just grabbed from OSDev or something? Now we should be able to use the reboot and/or shutdown functionality of syscon registers if they exist regardless of the SoC.

### CODE REVIEW

**Comment by kjhayes** [comment]
Don't merge this one until the Boot Allocator get's merged, it is a branch off of that PR and uses code from it.

File: drivers/syscon/syscon.c
Comment (egelja) [inline drivers/syscon/syscon.c]: this should be 0 not NULL

File: drivers/syscon/syscon.c
Comment (egelja) [inline drivers/syscon/syscon.c]: this should be 0 not NULL

File: kernel/of/fdt.c
Comment (egelja) [inline kernel/of/fdt.c]:
```suggestion
            if (fdt_prop_get_cell(phandle_prop, 0) == phandle) {
```

File: drivers/syscon/syscon.c
Comment (egelja) [inline drivers/syscon/syscon.c:91]: probably should check that `offset < syscon_size`

File: drivers/syscon/syscon.c
Comment (egelja) [inline drivers/syscon/syscon.c]:
```suggestion
        // All ones (no mask)
```

**Review by egelja — CHANGES_REQUESTED** [review]

**Review by egelja — APPROVED** [review]
lgtm

**Pull request merged**
